packages/web-components/src/donut-chart/donut-chart.ts

```diff
@@ -274,28 +293,28 @@ export class DonutChart extends FASTElement {
         }
         for (let j = 0; j < buttons.length; j++) {
           const legendRect = (buttons[j].getElementsByClassName('legendRect') as HTMLCollectionOf<HTMLDivElement>)[0];
-          legendRect.style['backgroundColor'] = legends[j].color;
+          legendRect.style['backgroundColor'] = legends![j].color;
 
           const legendText = (buttons[j].getElementsByClassName('legendText') as HTMLCollectionOf<HTMLDivElement>)[0];
           legendText.style['opacity'] = '1';
         }
       });
       buttons[i].addEventListener('click', () => {
-        if (this._selectedLegend === legends[i].title) {
+        if (this._selectedLegend === legends![i].title) {
```

Comment:
are these ! operators sufficient. In that case should the property be always required.
What happens if they are undefined.

Applies to all changes in this file.

Reply:
Here, `legends` will always be defined if the control enters the above for loop. I have also tried passing `undefined`, `[]` and `[{}]` as `chartData`, but none of these raised an error.
But I agree that it would be better to type-hint the required props for displaying the chart correctly.

Reply:
Can you create a task in the loop table to fix this.

Reply:
I have already pushed the changes to this PR